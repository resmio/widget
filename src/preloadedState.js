import moment from 'moment'
// import availabilities from './data/availabilities'

const preloadedState = {
  availabilities: [],
  BookingId: '',
  guestEmail: '',
  guestName: '',
  guestPhone: '',
  newsletterSubscription: false,
  maxGuests: 6,
  minGuests: 1,
  selectedDate: moment(),
  selectedGuests: 2,
  selectedAvailability: '',
  status: 'confirmed',
  timePeriodSelected: 0,
  timeFocused: '',
  buttonColor: '#3E4862',
  // Probably don't need the defaultHeight
  defaultHeight: '380px', // = 500px - 60px header - 60px footer
  defaultWidth: '330px',
  facility: 'meson-california-2',
  headerColor: '#53628C',
  headerImage: 'https://zenezake.files.wordpress.com/2015/07/img_6715.jpg',
  headerTextColor: 'white',
  numberOfPanels: 2,
  renderAtMaxSize: false,
  currentPanel: 2,
  guestSelectorState: 'collapsed',
  dateSelectorState: 'collapsed',
  timeSelectorState  : 'expanded',
  availabilitiesFetching: false,
  availabilitiesError: false
}

export default preloadedState
