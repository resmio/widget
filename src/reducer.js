function reducer(state = {}, action) {

  switch (action.type) {

    case 'GUEST_REMOVE':
      if (state.selectedGuests > state.minGuests) {
        return Object.assign({}, state, {
          selectedGuests: state.selectedGuests - 1
        })
      } else {
        return state
      }

    case 'GUEST_ADD':
      if (state.selectedGuests < state.maxGuests) {
        return Object.assign({}, state, {
          selectedGuests: state.selectedGuests + 1
        })
      } else {
        return state
      }

    case 'UI_GUEST_DROPDOWN_OPEN':
      return Object.assign({}, state, {
        guestSelectorCollapsed: false
      })

    case 'GUEST_SELECT':
      return Object.assign({}, state, {
        selectedGuests: parseInt(action.payload, 10) + 1,
        guestSelectorCollapsed: true
      })

    case 'UI_CALENDAR_SWITCH_FOCUS':
      return Object.assign({}, state, {
        calendarFocused: !state.calendarFocused
      })

    case 'DATE_SELECT':
      return Object.assign({}, state, {
        selectedDate: action.payload
      })

    case 'UI_PANEL_ADVANCE':
      if (state.currentPanel < state.numberOfPanels) {
        return Object.assign({}, state, {
          currentPanel: state.currentPanel + 1
        })
      } else {
        return state
      }

    case 'UI_PANEL_REDUCE':
      if (state.currentPanel >= 2) {
        return Object.assign({}, state, {
          currentPanel: state.currentPanel - 1
        })
      } else {
        return state
      }

    case 'BOOKING_POST':
      console.log('BOOKING_POST')
      return state

    default:
      return state
  }
}

export default reducer
