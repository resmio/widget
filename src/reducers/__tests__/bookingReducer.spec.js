import reducer from '../bookingReducer'
import {
  GUEST_ADD,
  GUEST_REMOVE,
  GUEST_SELECT
} from '../../actions/bookingActions'

import preloadedState from '../../preloadedState'

// We only need the booking part of the state
const state = preloadedState.booking

describe('booking reducer', () => {

  it('should add a guest to the inital state', () => {
    const action = { type: GUEST_ADD }
    const expected = 2
    const actual = reducer(state, action).selectedGuests

    expect(actual).toEqual(expected)
  })

  it('should remove a guest', () => {
    const testState = Object.assign(
      {}, state, {
        selectedGuests: 3
      }
    )

    const action = { type: GUEST_REMOVE }
    const expected = 2
    const actual = reducer(testState, action).selectedGuests

    expect(actual).toEqual(expected)
  })

  it('should not add a guest if it\'s already at maxGuests', () => {
    const testState = Object.assign(
      {}, state, {
        selectedGuests: 4,
        maxGuests: 4
      }
    )

    const action = { type: GUEST_ADD }

    const expected = 4
    const actual = reducer(testState, action).selectedGuests

    expect(actual).toEqual(expected)
  })

  it('should not remove a guest if it\'s already at min guests', () => {
    const testState = Object.assign(
      {}, state, {
        selectedGuests: 1
      }
    )
    const action = { type: GUEST_REMOVE }
    const expected = 1
    const actual = reducer(testState, action).selectedGuests

    expect(actual).toEqual(expected)
  })

  it('should select a specific number of guests', () => {
    const testState = Object.assign({}, state)
    const action = {
      type: GUEST_SELECT,
      payload: 4
    }
    const expected = 4
    const actual = reducer(testState, action).selectedGuests

    expect(actual).toEqual(expected)
  })

})
