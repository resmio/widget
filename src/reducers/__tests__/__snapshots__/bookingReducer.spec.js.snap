exports[`booking reducer should add a guest to the inital state 1`] = `2`;
